pkg/integrations/v2/eventhandler/eventhandler.go

```diff
@@ -0,0 +1,454 @@
+// Package eventhandler watches for Kubernetes Event objects and hands them off to
+// Agent's Logs subsystem (embedded promtail)
+package eventhandler
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/grafana/agent/pkg/integrations/v2"
+	"github.com/grafana/agent/pkg/logs"
+	"github.com/grafana/loki/clients/pkg/promtail/api"
+	"github.com/grafana/loki/pkg/logproto"
+	"github.com/prometheus/common/model"
+)
+
+const (
+	cacheFileMode = 0600
+)
+
+// EventHandler watches for Kubernetes Event objects and hands them off to
+// Agent's logs subsystem (embedded promtail).
+type EventHandler struct {
+	LokiClient    *logs.Instance
```

Comment:
Avoid storing a pointer to an instance: the instance can move to a new location in memory when a config changes, and the old pointer will be invalid (i.e., no longer running / sending logs). Instead, you should look up the instance from the subsystem any time you need to log something.